delta_shooter/game_mechanics.py

```diff
@@ -106,54 +123,63 @@ def __init__(
         else:
             self.screen = DummyScreen(GAME_SIZE)
 
-        self.reset()
         self.num_envs = 1
-        self.action_space = gym.spaces.Discrete(4)  # type: ignore
-
+        self.observation_space = gym.spaces.Box(low=-1, high=1, shape=(18,))
+        self.action_space = gym.spaces.Discrete(6)  # type: ignore
+        self.include_barriers = include_barriers
+        self.barriers = get_barriers() if include_barriers else []
+        self.reset()
         if self._render:
             self._draw()
 
     def reset(self) -> Tuple[np.ndarray, float, bool, Dict]:
         self.message = ""
-        spawn_idx = list(range(len(SPAWN_POINTS)))
-        random.shuffle(spawn_idx)
 
-        player1_idx = spawn_idx.pop()
-        player2_idx = spawn_idx.pop()
+        opposite_spawn = {0: 1, 1: 0, 2: 3, 3: 2}
+        player1_idx = random.choice(range(4))
+        player2_idx = opposite_spawn[player1_idx]
```

Comment:
Is this because we don't want the opposition to be opposite us? Think a comment might help to explain this :smile: